### ISSUE TITLE
Negative Temperatures are not read correctly

### ISSUE THREAD

**Issue by schmiegelt** [schmiegelt/Py-Weishaupt-WCM-COM#1]
Negative Temepratures are read as a very high positive temperature.

**Comment by add30417** [comment]
Right, how can I recalculate the actually value? Do you have any solution or hint?

**Comment by add30417** [comment]
var usValue;
    /* check: positive value */
    if (HighByte <= 127) {
        /* positive value */
        usValue = (HighByte * 256) + LowByte;
    }
    /* check: replacement value? */
    else if ((HighByte === 128) && (LowByte === 0)) {
        /* default value */

        //positve value (default value)
        usValue = (HighByte * 256) + LowByte;
    } else {
        /* negativ value */
        usValue = ((-32768) + ((HighByte - 128) * 256) + LowByte);
    }
    return usValue;

**Comment by add30417** [comment]
#3 I found a solution, maybe you can accept my Pull requests.

**Comment by erwinm9m** [comment]
convert as follows

def int_from_bytes(b):
    if not b: # special-case 0 to avoid b[0] raising
        return 0
    n = b[0] & 0x7f # skip sign bit
    for by in b[1:]:
        n = n * 256 + by
    if b[0] & 0x80: # if sign bit is set, 2's complement
        bits = 8*len(b)
        offset = 2**(bits-1)
        return n - offset
    else:
        return n

def getTemperature(lowByte, highByte):
    return 0.1*int_from_bytes(bytearray([highByte,lowByte]))
